src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -441,44 +506,323 @@ public int getID()
         {
-            this.triggerAchievement(ModAchievements.achievementKillOneMob);
+            case TAG_GUARD_MOBS:
+                return killedMobs;
+            case TAG_MINER_ORES:
+                return minedOres;
+            case TAG_MINER_DIAMONDS:
+                return minedDiamonds;
+            case TAG_BUILDER_HUTS:
+                return builtHuts;
+            case TAG_FISHERMAN_FISH:
+                return caughtFish;
+            case TAG_FARMER_WHEAT:
+                return harvestedWheat;
+            case TAG_FARMER_POTATOES:
+                return harvestedPotatoes;
+            case TAG_FARMER_CARROTS:
+                return harvestedCarrots;
+            case TAG_LUMBERJACK_SAPLINGS:
+                return plantedSaplings;
+            case TAG_LUMBERJACK_TREES:
+                return felledTrees;
+            default:
+                return 0;
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_SECOND)
+    }
+
+    /**
+     * Trigger first achievement.
+     * @param statistic the statistic.
+     */
+    private void triggerFirstAchievement(@NotNull String statistic)
+    {
+        switch (statistic)
         {
-            this.triggerAchievement(ModAchievements.achievementKill25Mobs);
+            case TAG_GUARD_MOBS:
+                this.triggerAchievement(ModAchievements.achievementKillOneMob);
+                break;
+            case TAG_MINER_ORES:
+                this.triggerAchievement(ModAchievements.achievementMineOneOre);
+                break;
+            case TAG_MINER_DIAMONDS:
+                this.triggerAchievement(ModAchievements.achievementMineOneDiamond);
+                break;
+            case TAG_BUILDER_HUTS:
+                this.triggerAchievement(ModAchievements.achievementBuildOneHut);
+                break;
+            case TAG_FISHERMAN_FISH:
+                this.triggerAchievement(ModAchievements.achievementCatchOneFish);
+                break;
+            case TAG_FARMER_WHEAT:
+                this.triggerAchievement(ModAchievements.achievementHarvestOneWheat);
+                break;
+            case TAG_FARMER_POTATOES:
+                this.triggerAchievement(ModAchievements.achievementHarvestOnePotato);
+                break;
+            case TAG_FARMER_CARROTS:
+                this.triggerAchievement(ModAchievements.achievementHarvestOneCarrot);
+                break;
+            case TAG_LUMBERJACK_SAPLINGS:
+                this.triggerAchievement(ModAchievements.achievementPlantOneSapling);
+                break;
+            case TAG_LUMBERJACK_TREES:
+                this.triggerAchievement(ModAchievements.achievementFellOneTree);
+                break;
+            default:
+                break;
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_THIRD)
+    }
+    /**
+     * Trigger second achievement.
+     * @param statistic the statistic.
+     */
+    private void triggerSecondAchievement(@NotNull String statistic)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "triggerSecondAchievement" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)